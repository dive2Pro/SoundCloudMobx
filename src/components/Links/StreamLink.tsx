import * as React from 'react'
import { NavLink } from 'react-router-dom'

interface IStreamLinkProp {
  id: number,
  children?: any
  className?:string
}

const StreamLink = (props: IStreamLinkProp) => {
  const { id, children, ...rest } = props
  return (
    <NavLink
      to={{ pathname: '/stream', search: `?id=${id}` }}
      {...rest}
    >
      {children}
    </NavLink >
  )
}

export default StreamLink