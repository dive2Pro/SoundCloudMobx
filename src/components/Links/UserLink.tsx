import * as React from 'react'
import { NavLink as Link } from 'react-router-dom'
// const styles = require('./.scss')

interface IRouterLinkProp {
  path: string,
  id: number
  children?: any
  className?: string
}

const RouterLink = (prop: IRouterLinkProp) => {

  return (
    <Link
      to={{
        pathname: `/users/${prop.path}`,
        search: `?id=${prop.id}`
      }}
    >
      {prop.children}
    </Link>
  );

}

export default (RouterLink);