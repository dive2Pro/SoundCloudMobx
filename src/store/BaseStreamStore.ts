import performanceStore from './PerformanceStore';
import {
  action, observable, runInAction
  , computed, ObservableMap, autorun
  , IReactionDisposer, IObservableArray
} from 'mobx';

import {
  ITrack
} from '../interfaces/interface';
import { GENRES } from '../constants/trackTypes'

export interface IBaseActStore {
  isLoading: boolean;
  currentGenre: string
  isError: (genre: string) => boolean
}
abstract class BaseStreamStore<T> implements IBaseActStore {
  static displayName = 'BaseAct'
  // 记得初始化
  protected itemsMap = new ObservableMap<T[]>();
  @observable protected nextHrefsByGenre = new ObservableMap<string>();
  protected isLoadingByGenre = {
    get: performanceStore.getLoadingStateWidthKey
  }
  protected isErrorsMap = new ObservableMap<boolean>()

  // 排序和搜索过滤
  @observable filterType: string
  @observable filterTitle: string
  @observable sortType: string
  @observable filteredTracks: ITrack[] = []
  @observable currentGenre: string = GENRES[0]

  autorunHandle: IReactionDisposer;

  @computed get isLoading(): boolean {
    return this.isLoadingByGenre.get(this.currentGenre)
  }
  
  @computed get nextHref() {
    return this.nextHrefsByGenre.get(this.currentGenre) || ''
  }

  @action setFilterTitle(title: string) {
    this.filterTitle = title;
  }

  @action setSortType(type: string) {
    this.sortType = type;
  }

  @action setFilterType(filterType: string = '') {
    this.filterType = filterType;
  }

  setLoadingByGenre(genre: string, loading: boolean) {
    // this.isLoadingByGenre.set(genre, loading);
    performanceStore.setLoadingStateWithKey(genre, loading);
  }

  @action setNextHrefByGenre(genre: string, nextHref: string) {
    this.nextHrefsByGenre.set(genre, nextHref)
  }

  @action setGenre(genre: string) {
    // genre = genre.toLocaleLowerCase();
    this.currentGenre = genre;
    performanceStore.setCurrentGenre(this.currentGenre)
    this.initFilterFunction(genre);
  }

  @computed get currentItems(): T[] {
    const hasItems = this.itemsMap.has(this.currentGenre)
    if (!hasItems) {
      this.itemsMap.set(this.currentGenre,[])
    }
    const item = this.itemsMap.get(this.currentGenre);
    return <T[]>item
  }

  isError = (genre: string): boolean => {
    return performanceStore.isError(genre)
  }

  @action protected catchErr = (err: any, genre: string) => {
    performanceStore.catchErr(err, genre)
    console.error(err, genre)
  }
  protected resetErrorWithType = (type: string) => {
    performanceStore.resetErrorsMap(type)
  }
  protected filterByFilterType(fs: T[]): T[] {
    return fs;
  }
  abstract transToTracks(act: T[]): ITrack[];

  private async filterFunc(activities: T[], filterTitle: string, sortType: string, filterType: string) {
    let fs: ITrack[] = []
    const filterByTypes = await this.filterByFilterType(activities)
    // console.log(filterByTypes)
    fs = await this.transToTracks(filterByTypes)
    fs = await this.filterByFilterTitle(fs);
    fs = await this.filterBySortType(fs);

    runInAction('set-filteredTracks', () => {
      (<IObservableArray<ITrack>>this.filteredTracks).replace(fs)
    })
  }
  private filterBySortType(fs: ITrack[]) {
    let temp = fs.slice();
    if (!!this.sortType) {
      temp = fs.sort((p, n) => {
        let pCount = p[this.sortType];
        let nCount = n[this.sortType];
        pCount = !Number.isNaN(pCount) ? pCount : 0;
        nCount = !Number.isNaN(nCount) ? nCount : 0;
        return nCount - pCount;
      })
    }
    return temp
  }
  private filterByFilterTitle(fs: ITrack[]) {
    let temp = fs.slice();
    if (!!this.filterTitle) {
      temp = fs.filter(item => {
        return item.title.indexOf(this.filterTitle) > -1
      })
    }
    return temp
  }


  private initFilterFunction(type: string) {

    if (this.autorunHandle) {
      this.autorunHandle();
    }
    let has = this.itemsMap.has(type);
    if (!has) {
      this.itemsMap.set(type, []);
    }
    this.autorunHandle = autorun(() => {
      this.filterFunc(<T[]>this.itemsMap.get(type), this.filterTitle, this.sortType, this.filterType)
    })
  } 
}
export default BaseStreamStore