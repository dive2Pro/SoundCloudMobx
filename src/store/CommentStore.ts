import {
  ObservableMap, observable, computed, action, runInAction
  // , createTransformer
} from 'mobx'
import { apiUrl, unauthApiUrl } from '../services/soundcloundApi'
import { ITrack } from '../interfaces/interface';
import  performanceStore  from './PerformanceStore'
import  sessionStore  from './SessionStore'
import { FETCH_COMMENTS } from '../constants/fetchTypes';
import { POST_TARCK_COMMENT } from '../constants/requestTypes';

import { RaceFetch as fetch } from '../services/Fetch'

export interface IComment {
  id: number,
  created_at: string,
  user_id: number,
  track_id: number,
  timestamp: number,
  body: string
  uri: string,
  user: {
    id: number,
    permalink: string,
    username: string,
    uri: string,
    permalink_url: string,
    avatar_url: string
  }
}



export class CommentStore {
  @observable currentTrack: ITrack|null = null
  @observable private nextHrefsByTrack
   = new ObservableMap<string>();
  
  private commentsByTracks = new ObservableMap<IComment[]>();

  constructor() {
    performanceStore.setLoadingStateWithKey(FETCH_COMMENTS, false)
    performanceStore.setLoadingStateWithKey(POST_TARCK_COMMENT, false)
  }
 

  


  @computed get isLoadingMoreComment() { 
    return performanceStore.getLoadingStateWidthKey(FETCH_COMMENTS);
  }

  @computed get commentsCount() {
    return this.currentTrackComments.length
  }

  @computed get currentTrackComments() {
    if (!this.currentTrack) { return []; }
    const id =this.currentTrack.id + ''
    const has = this.commentsByTracks.has(id)
    if (!has) {
      this.commentsByTracks.set(id,[])
    }
    return this.commentsByTracks.get(id)||[]
  }
  @computed get currentCommentNextHref(): string {
    return this.currentTrack &&this.nextHrefsByTrack.get(this.currentTrack.id + '') || ''
  }

  setLoadingState(type: string, loading: boolean) {
    performanceStore.setLoadingStateWithKey(type, loading);
  }

  @action setCurrentTrack(track: ITrack) {
    this.currentTrack = (track);
    
  }

  @action async fetchMoreComments(nextHref?: string) {
    if (this.isLoadingMoreComment) { return }
    nextHref = this.currentCommentNextHref
    if (nextHref == 'EMPTY'||!this.currentTrack) {
      return
    }
    const { id } = this.currentTrack
    const keyId = id + ''
    let url = nextHref
      ? nextHref : apiUrl(`tracks/${id}/comments?linked_partitioning=1&limit=50&offset=0`, '&');

    this.setLoadingState(FETCH_COMMENTS, true)

    try {
       const data: any = await fetch(url); 
       const oldData = this.commentsByTracks.get(keyId)
       runInAction(() => {
        if (oldData) {
          oldData.push(...data.collection);
        } else {
          this.commentsByTracks.set(keyId, data.collection);
        }
        this.nextHrefsByTrack.set(keyId, data.next_href || 'EMPTY');
      })
    } catch (err) {
      performanceStore.catchErr(err, FETCH_COMMENTS)
    } finally {
      this.setLoadingState(FETCH_COMMENTS, false)
    }
  }

  // todo 用decortator来处理 try_catch  
  async submitReplay(id: number, msg: string) {
    const user = sessionStore.user
    if (!user) {
      return
    } 
    
    const url = unauthApiUrl(`/tracks/${id}/comments`, '&')
    try {
      this.setLoadingState(POST_TARCK_COMMENT, true)
      // const data = await fetch(url, {
      // method: 'POST', 'Content-Type': `application/x-www-form-urlencoded; charset=UTF-8 `
      // })      
      const data = await new Promise((resolve, reject) => {
        setTimeout(() => {
          resolve(rawData(msg))
         
        }, 500)
      }) 
      this.pushLastReplay(data)
      
    } catch (err) {
      performanceStore.catchErr(err, POST_TARCK_COMMENT)
    } finally {
      this.setLoadingState(POST_TARCK_COMMENT, false)
    }

  }
  @action private pushLastReplay(data) {
    if (!this.currentTrack) { 
      return
    }
    const comments = this.currentTrackComments;  
    comments.unshift(data)
  }
}
  export const rawData= (msg:string):IComment => ({
            kind: 'comment'
            , id: 341101782,
            created_at: '2017/04/13 16:23:38 +0000',
            user_id: 278227204,
            track_id: 280186218,
            timestamp: 450,
            body: `@huang-alex-635833920: ${msg}`,
            uri: 'https://api.soundcloud.com/comments/341101782',
            user: {
              id: 278227204
              , kind: 'user',
              permalink: 'huang-alex-635833920',
              username: 'Huang Alex',
              last_modified: '2017/03/31 08:35:22 +0000',
              uri: 'https://api.soundcloud.com/users/278227204',
              permalink_url: 'http://soundcloud.com/huang-alex-635833920',
              avatar_url: 'https://i1.sndcdn.com/avatars-000286355267-ibfmck-large.jpg'
            }
  })
          
export default new CommentStore();