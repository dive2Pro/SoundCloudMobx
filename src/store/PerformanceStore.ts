import {
  observable
  , action
  , IObservableArray
  , ObservableMap
  , computed
  , expr
} from 'mobx'

export class PerformanceStore {

   __breaks={
    $breakMedium:700
  }

  @observable windowWidth = window.innerWidth;

  onceLoadingIsAllSettle: boolean;
  scrollLimitByGenre = new ObservableMap<number[]>()
  @observable genre: string
  // 这个用来记录当前应该被 player 进行毛玻璃效果处理的dom id
  @observable glassNode: string
  @observable scrollY: number
  isLoadingsByKey = new ObservableMap<boolean>();
  isErrorsMap = new ObservableMap<boolean>()
  
  @computed get scrollLimit(): number[] {
    return this.scrollLimitByGenre.get(this.genre) || [];
  }

  @action setCurrentGenre(genre: string) {
    this.genre = genre;
    if (!this.scrollLimitByGenre.get(genre)) {
      this.scrollLimitByGenre.set(
        genre,
        [window.innerHeight, window.innerHeight]
      )
    }
  }

  getLoadingState(type: string): boolean {
    return this.isLoadingsByKey.get(type) || false
  }

  @action setScrollLimit(...limit: number[]) {
    const map = this.scrollLimitByGenre.get(this.genre);
    if (map) {
      (<IObservableArray<number>>map).replace(limit);
    } else {
      this.scrollLimitByGenre.set(this.genre, limit)
    }
  }

  @action setCurrentGlassNodeId(id: string) {
    this.glassNode = id;
  }

  @action setScrollY(y: number) {
    this.scrollY = y;
  }

  @action setLoadingStateWithKey = (key: string, loading: boolean) => {
    this.isLoadingsByKey.set(key, loading)
  }

  getLoadingStateWidthKey = (key: string) => {
    return this.isLoadingsByKey.get(key) || false
  }

  /**
   * 当前所以请求完毕
   */
  @computed get allLoadingIsSettle(): boolean {
    const allSettle = expr(() =>
      Array.from(this.isLoadingsByKey.values()).every(v => v === false))
    return allSettle
  }


  isError = (genre: string): boolean => {
    return this.isErrorsMap.get(genre) || false
  }

  @action catchErr = (err: any, genre: string) => {
    this.isErrorsMap.set(genre, true);
  }

  @action resetErrorsMap = (fetchType: string) => {
    this.isErrorsMap.set(fetchType, false);
  }

  @action setWindowSize(size){
    this.windowWidth = size
  }

  @computed get isUnderLarge(){
    return this.windowWidth < 1200
  }

  @computed get iUnderMedium(){
    return this.windowWidth < 680
  }

  @computed get iUnderHandsets(){
    return this.windowWidth < 475
  }

}

export default new PerformanceStore()