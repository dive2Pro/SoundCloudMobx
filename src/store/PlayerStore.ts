import { observable, action, computed } from 'mobx';
import { addClientId } from '../services/soundcloundApi'
import { ITrack } from '../interfaces/interface'

const isEqual =require('lodash/isEqual')
export class PlayerStore {
  @observable playingTrack: ITrack
  @observable isPlaying: boolean = false;
  @observable playList: ITrack[] = [];

  @observable isPlaylistOpen: boolean = false;
  @observable isShuffleMode: boolean = false;
  @observable isVolumeOpen: boolean = false

  @observable volume: number = 0.25

  @action setVolume(v: number) {
    this.volume = v;
  }

  @action setPlayingTrack(track: ITrack | number) {
    if (typeof track === 'number') {
      track = this.playList[track];
    }

    if (isEqual(this.playingTrack, track) && this.isPlaying) {
      this.isPlaying = false;
      
    } else {
      this.playingTrack = track;
      this.isPlaying = true;
      this.addToPlaylist(track);
    }
  }

  @action togglePlaying() {
    if(!this.playingTrack){
        this.playNextTrack(1)
    }else{
      this.isPlaying = !this.isPlaying;

    }

  }
  @action toggleVolumeOpen(open?: boolean) {
    if (open != null) {
      this.isVolumeOpen = open
    } else {
      this.isVolumeOpen = !this.isVolumeOpen;
    }
  }

  @action toggleShuffleMode() {
    this.isShuffleMode = !this.isShuffleMode
  }


  @action clearPlaylist() {
    this.playList = [];
  }
  @action removeFromPlaylist(...tracks: ITrack[]) {
    tracks.forEach(t => {
      const index = this.playList.indexOf(t)
      this.playList.splice(index, 1)
    })
  }

  @action togglePlaylistOpen(open?: boolean) {
    if (open != null) {
      this.isPlaylistOpen = open
    } else {
      this.isPlaylistOpen = !this.isPlaylistOpen;
    }
  }

  @computed get getPlaylistCount() {
    return this.playList && this.playList.length;
  }
  @computed get playingUrl() {
    if (this.playingTrack) {
      let url = this.playingTrack.uri + '/stream';
      url = addClientId(url, '?');
      return url
    }
    return ''
  }
  @action playNextTrack(diff: number): boolean {
    if (this.getPlaylistCount < 1) {
      return false;
    }
    
    let nextTrack: ITrack | undefined;
    if (this.isShuffleMode) {
      const newList = this.playList.slice();
      const deleIndex = newList.indexOf(this.playingTrack)
      newList.splice(deleIndex, 1);
      // tslint:disable-next-line:radix
      nextTrack = newList[parseInt(Math.random() * 10 + '') % newList.length];
    } else {
      let changedIndex = 0;
      if (this.playingTrack) {
        changedIndex = this.playList.indexOf(this.playingTrack) + diff;
      }

      nextTrack = this.playList[changedIndex];
    }
    if (nextTrack != null) {
      this.setPlayingTrack(nextTrack);
      return true
    }

    return false
  }

  addToPlaylist(tracks: ITrack | ITrack[]) {
    let ts:any = tracks
    if (ts.slice&&Array.isArray(ts.slice())) {
      (<ITrack[]>tracks).slice().forEach((t, i) => {
        i === 0 && this.setPlayingTrack(t)
        this.pushToPlayerlist(t)
      })
    } else {
      this.pushToPlayerlist(<ITrack>tracks)
    }
  }

  @action private pushToPlayerlist(t: ITrack) {
    if (!this.playList.some((item)=>item.id===t.id)) {
      this.playList.push(t)
    }
  }
}

export default new PlayerStore();