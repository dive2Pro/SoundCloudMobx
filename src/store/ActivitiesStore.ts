import { action, computed, runInAction } from 'mobx'
import  BaseStreamStore  from './BaseStreamStore';
import { IActivitiesItem, ITrack } from '../interfaces/interface';
import { FETCH_ACTIVITIES } from '../constants/fetchTypes';
import  performanceStore from './PerformanceStore';
import { addAccessToken, apiUrl } from '../services/soundcloundApi';
import { RaceFetch as fetch } from '../services/Fetch'

export class ActivitiesStore extends BaseStreamStore<IActivitiesItem>  {

  constructor() {
    super()
    this.setGenre(FETCH_ACTIVITIES)
  }

  @action setNextActivitiesHref(nextHref: string) {
    this.setNextHrefByGenre(FETCH_ACTIVITIES, nextHref)
  }

  @action addActivities(arr: IActivitiesItem[]) {
    const itemsLength = this.currentItems.length;
    this.currentItems.splice(itemsLength, 0, ...arr);
    // const t : any= []
    // const iis:any = t.splice(0, 0, ...arr)
    // console.log(
    //   // t, '-------------',
    //   this.currentItems)
  }

  transToTracks(items: IActivitiesItem[]): ITrack[] {
    return items.map(this.getAllTrackFromActivity)
      .filter(item => item != null);
  }

  getAllTrackFromActivity(act: IActivitiesItem) {
    return act.origin
  }

  @computed get tracks() {
    return this.currentItems && this.currentItems.map(this.getAllTrackFromActivity)
  }

  filterByFilterType(fs: IActivitiesItem[]) {
    let temp = fs.slice();
    if (!!this.filterType) {
      temp = fs.filter(item => {
        return item.type === this.filterType
      })
    }
    return temp;
  }
  async filterActivities(arr: IActivitiesItem[]) {
    const filterArr = await arr.filter(item => {
    const b = this.currentItems.some(active =>
        active.created_at === item.created_at)
    return !b;
    })
    this.addActivities(filterArr);
  }

  setLoadingActivities(b: boolean) {
    this.setLoadingByGenre(FETCH_ACTIVITIES, b);
  }

  async fetchNextActivities(first?: boolean) {
     
    if (this.isLoading) {
       
      return  
    }
    
    if (first && this.currentItems.length > 0) {
      return
    }

    performanceStore.setCurrentGenre(this.currentGenre)
    await this.fetchActivities(this.nextHref);
  }
  @action private async  fetchActivities(nextHref?: string) {
    if (this.isLoading) { return }
    let activitiesUrl = nextHref ? addAccessToken(nextHref, '&') :
      apiUrl(`me/activities?limit=50`, '&')
    try {
      this.setLoadingActivities(true);
      const data: any = await fetch(activitiesUrl)
      runInAction(() => {
        this.setNextActivitiesHref(data.next_href)
        this.filterActivities(data.collection);
      })
    } catch (err) {
      this.catchErr(err, this.currentGenre)
    }
    finally {
      this.setLoadingActivities(false)
    }
  }
}

const activitiesStore = new ActivitiesStore();

export { activitiesStore as default };
