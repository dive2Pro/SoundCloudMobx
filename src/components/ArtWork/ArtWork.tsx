import * as React from 'react'
const preImage = require('../../../public/images/preload.jpg')

import { autorun, IReactionDisposer } from 'mobx'
import {
  observer
  , inject
} from 'mobx-react'
import { PerformanceStore } from '../../store/PerformanceStore';
import { PERFORMANCE_STORE } from '../../constants/storeTypes';

export interface IArtWorkProps {
  size?: number
  alt?: string
  src: string
  optionalImg?: string,
  clazz?: string
  style?: { width?: any, height?: any }
  performanceStore?: PerformanceStore
  live?: boolean
  onClick?: (e: any) => void
}

@inject(PERFORMANCE_STORE)
@observer
class ArtWork extends React.Component<IArtWorkProps, any> {
  loadImage: any;

  handlerObserver: IReactionDisposer;
  img: HTMLImageElement;
  retryCount = 3
  constructor() {
    super()
    this.loadImage = new Image()
  }
  componentDidMount() {
    this.loadImage.onload = () => {
      if (this.img) {
        this.img.src = this.loadImage.src
        // 取消观察
        if (!this.props.live) {
          this.handlerObserver()
        }
      }
    }

    this.loadImage.onerror = () => {
      // 出现错误怎么办?  
      if (this.retryCount-- > 0) {
        this.caclImg(this.props.src)
      } else {
        // show broken?
      }
    }

    this.handlerObserver = autorun(() => {
      const { performanceStore: ps, src } = this.props
      if (this.img && ps && ps.scrollLimit.length > 0 && src) {
        const [l, h] = ps.scrollLimit
        const y = this.img.y
        const imgHeight = this.img.offsetHeight;
        if ((l === h && y <= l) || (y < l && (y + imgHeight > h))) {
          this.caclImg(src)
        }
      } else if (this.img && !!src === false) {
        this.img.src = preImage
      }
    })
  }
  componentWillUpdate(nextProps, nextState) {
    if (nextProps.src !== this.img.src) {

      this.caclImg(nextProps.src)
    }
  }
  caclImg = (src: string) => {
    const reg = /-{1}large\.{1}/
    let { size, style } = this.props

    if (src && (size && size <= 50)) {
      src = src.replace(reg, '-badge\.')
    } else if (src && style) {
      // tslint:disable-next-line:radix
      const width = parseInt(style.width)
      const height = parseInt(style.height);

      if (width > 100 || height > 100) {
        const size = Math.max(Math.ceil(width), Math.ceil(height))
        let replace = 't300x300'
        // console.log(size)
        if (size > 300 && size < 500) {
          replace = 'crop'
        } else {
          replace = 't500x500'
        }
        src = src.replace(reg, `-${replace}\.`);
      }
    }
    this.loadImage.src = src

  }
  render() {
    const
      {
         size
        , clazz
        , alt
        , style
        , src, onClick } = this.props
    return (
      <img
        ref={n => this.img = n}
        className={clazz}
        src={src ? preImage : preImage}
        width={size}
        height={size}
        alt={alt}
        style={style}
        onClick={onClick}
      />
    )
  }
}
export default (ArtWork);