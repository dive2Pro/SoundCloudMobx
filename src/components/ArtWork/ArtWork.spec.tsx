import * as React from 'react'
import ArtWork from './index'
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import { StreamLink } from '../Links'
import { Action } from '../HoverActions'
import { performanceStore } from '../../store';
import { Provider } from 'mobx-react'

describe('ArtWork Component', () => {
  it('初始化渲染', () => {
    const track: any = { title: 'Test---', user: { username: 'haha' } }
    const props = {
      track: track,
      src: 'http://img4.imgtn.bdimg.com/it/u=1350614941,725003865&fm=23&gp=0.jpg'
    }
    const wrapper = shallow(
      <Provider performanceStore={performanceStore}>
        <ArtWork {...props} />
      </Provider>
    );
    expect(toJson(wrapper)).toMatchSnapshot()
    const imgs = wrapper.find('img')
    expect(imgs.length).toBe(0)
    const img = imgs.first()

    // expect(img).toHaveProperty("src", props.src);
  })
})