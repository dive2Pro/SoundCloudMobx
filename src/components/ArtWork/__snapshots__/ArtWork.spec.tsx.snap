exports[`ArtWork Component 初始化渲染 1`] = `
<inject-ArtWork-with-performanceStore
  src="http://img4.imgtn.bdimg.com/it/u=1350614941,725003865&fm=23&gp=0.jpg"
  track={
    Object {
      "title": "Test---",
      "user": Object {
        "username": "haha",
      },
    }
  } />
`;
